astrowidgets/core.py

```diff
@@ -619,6 +619,32 @@ def _validate_marker_name(self, marker_name):
                              '{}'.format(marker_name,
                                          ', '.join(RESERVED_MARKER_SET_NAMES)))
 
+    def add_regions(self, regions, marker_name='_astropy_regions'):
+        """
+        Creates markers in the image from given regions from
+        the Astropy ``regions`` package.
+
+        .. todo::
+
+            How to tied this into existing marker machinery?
+
+        Parameters
+        ----------
+        regions : list of obj
+            ``regions`` shapes.
+
+        marker_name : str
+            Name to assign the markers created from regions.
+
+        """
+        from ginga.util import ap_region
+
+        # Regions will be appended if other regions were previously
+        # added in another call.
+        # TODO: How to get or set the canvas name?
+        for reg in regions:
+            ap_region.add_region(self._viewer.canvas, reg, tag=marker_name)
```

Comment:
This fails for all regions after the first the error below. We get around this in `add_markers` by creating a compound object. We could (maybe?) do that here, or generate a name for each region or something like that.


```python traceback
---------------------------------------------------------------------------
CanvasError                               Traceback (most recent call last)
<ipython-input-10-a84f619487a4> in <module>
----> 1 w.add_regions(ds9_pixel_regions)

~/development/astronomy/astrowidgets/astrowidgets/core.py in add_regions(self, regions, marker_name)
    644         # TODO: How to get or set the canvas name?
    645         for reg in regions:
--> 646             ap_region.add_region(self._viewer.canvas, reg, tag=marker_name)
    647 
    648     def add_markers(self, table, x_colname='x', y_colname='y',

~/miniconda3/envs/awid-ging-dev/lib/python3.7/site-packages/ginga/util/ap_region.py in add_region(canvas, r, tag, redraw)
    164         tag = obj.get_data('name')
    165     if obj is not None:
--> 166         canvas.add(obj, tag=tag, redraw=redraw)
    167         return obj
    168 

~/miniconda3/envs/awid-ging-dev/lib/python3.7/site-packages/ginga/canvas/CanvasMixin.py in add(self, obj, tag, tagpfx, belowThis, redraw)
     51             # user supplied a tag
     52             if tag in self.tags:
---> 53                 raise CanvasError("Tag already used: '%s'" % (tag))
     54         else:
     55             if tagpfx:

CanvasError: Tag already used: '_astropy_regions'
```

Reply:
This is where we handle tag names in markers: https://github.com/astropy/astrowidgets/pull/96/files#diff-47161bfaadb2c20c9c4f939f18fb08aaL697-L710